### ISSUE TITLE
unable to run any nannou code on Ubuntu 20.04

### ISSUE THREAD

**Issue by schwaaweb** [nannou-org/nannou#585]
Hello,

I am running Ubuntu 20.04 (I was successfully able to run the demos on Ubuntu 19.10 and OSX)

Any guidance for next steps would be appreciated!

Thanks
- Darwin

 $ uname -a                                                                                                                                                                                                     
Linux schwaarocm2004 5.4.0-28-lowlatency #32-Ubuntu SMP PREEMPT Wed Apr 22 18:35:27 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux

- Installation:

sudo apt-get install curl build-essential python cmake pkg-config
sudo apt-get install libasound2-dev
sudo apt install libxcb-shape0-dev libxcb-xfixes0-dev libx11-dev
sudo apt-get install nvidia-settings nvidia-driver-440 vulkan-utils
git clone https://github.com/nannou-org/nannou.git && cd nannou
cargo check
cargo build
RUST_BACKTRACE=full cargo run --release --example draw

                 ...

   Compiling time_calc v0.13.0
   Compiling rustc_version v0.2.3
   Compiling parking_lot_core v0.6.2
   Compiling parking_lot v0.9.0
   Compiling isf v0.1.0 (https://github.com/nannou-org/isf#99424e5b)
   Compiling conrod_core v0.69.0
   Compiling wgpu-native v0.4.3
   Compiling nannou_timeline v0.14.0 (/home/schwaahed/work/nannou/nannou_timeline)
   Compiling wgpu v0.4.0
   Compiling conrod_wgpu v0.69.0
   Compiling nannou v0.14.0 (/home/schwaahed/work/nannou/nannou)
warning: unused return value of `std::mem::replace` that must be used
   --> nannou/src/draw/mod.rs:171:9
    |
171 |         mem::replace(&mut self.drawing, drawing);
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: `#[warn(unused_must_use)]` on by default
    = note: if you don't need the old value, you can just assign the new value directly

   Compiling nannou_isf v0.1.0 (/home/schwaahed/work/nannou/nannou_isf)
warning: 1 warning emitted

   Compiling examples v0.1.0 (/home/schwaahed/work/nannou/examples)
    Finished release [optimized] target(s) in 2m 29s
     Running `target/release/examples/draw`
thread 'main' panicked at 'called `Option::unwrap()` on a `None` value', /home/schwaahed/.cargo/registry/src/github.com-1ecc6299db9ec823/wgpu-native-0.4.3/src/instance.rs:474:5
stack backtrace:
   0:     0x55eaf1142944 - backtrace::backtrace::libunwind::trace::hfcd33194db0151d4
                               at /cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.46/src/backtrace/libunwind.rs:86
   1:     0x55eaf1142944 - backtrace::backtrace::trace_unsynchronized::hfd1904bbbd5335b5
                               at /cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.46/src/backtrace/mod.rs:66
   2:     0x55eaf1142944 - std::sys_common::backtrace::_print_fmt::h8476c57b177b254e
                               at src/libstd/sys_common/backtrace.rs:78
   3:     0x55eaf1142944 - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h73acbc5f6d4b1044
                               at src/libstd/sys_common/backtrace.rs:59
   4:     0x55eaf116a2ac - core::fmt::write::hdf236390fbd68d3d
                               at src/libcore/fmt/mod.rs:1069
   5:     0x55eaf113f833 - std::io::Write::write_fmt::h5722fa40bb2afafd
                               at src/libstd/io/mod.rs:1532
   6:     0x55eaf1145175 - std::sys_common::backtrace::_print::ha468e873aada7c78
                               at src/libstd/sys_common/backtrace.rs:62
   7:     0x55eaf1145175 - std::sys_common::backtrace::print::h149365a2f029de62
                               at src/libstd/sys_common/backtrace.rs:49
   8:     0x55eaf1145175 - std::panicking::default_hook::{{closure}}::hb4a33f9e05934a52
                               at src/libstd/panicking.rs:198
   9:     0x55eaf1144eb2 - std::panicking::default_hook::hc4535d7b0c743abd
                               at src/libstd/panicking.rs:218
  10:     0x55eaf11457d2 - std::panicking::rust_panic_with_hook::haa34a96a6dbd5a2e
                               at src/libstd/panicking.rs:477
  11:     0x55eaf11453bb - rust_begin_unwind
                               at src/libstd/panicking.rs:385
  12:     0x55eaf1168071 - core::panicking::panic_fmt::hd101a87121fa411f
                               at src/libcore/panicking.rs:89
  13:     0x55eaf1167fbd - core::panicking::panic::h02171c407fa1462f
                               at src/libcore/panicking.rs:52
  14:     0x55eaf105bbcf - wgpu_request_adapter
  15:     0x55eaf0ffe157 - wgpu::Adapter::request::h62fd56fea81fe726
  16:     0x55eaf0f082b6 - nannou::wgpu::device_map::AdapterMap::get_or_request::h4a0c880c96a8469f
  17:     0x55eaf0f34628 - nannou::window::Builder::build::hf8a5ed80233647f6
  18:     0x55eaf0ee5268 - nannou::app::SketchBuilder<E>::run::h8163d2e53b384d8a
  19:     0x55eaf0ebdd40 - draw::main::h280daa4a981895c3
  20:     0x55eaf0eeb373 - std::rt::lang_start::{{closure}}::hef45f9ef9cb54194
  21:     0x55eaf1145ce8 - std::rt::lang_start_internal::{{closure}}::h8ee41efd8ae7ff76
                               at src/libstd/rt.rs:52
  22:     0x55eaf1145ce8 - std::panicking::try::do_call::h0a4c4514ee6ebbfc
                               at src/libstd/panicking.rs:297
  23:     0x55eaf1145ce8 - std::panicking::try::h7c2f3858e42d7516
                               at src/libstd/panicking.rs:274
  24:     0x55eaf1145ce8 - std::panic::catch_unwind::h76c4030fa5655c1e
                               at src/libstd/panic.rs:394
  25:     0x55eaf1145ce8 - std::rt::lang_start_internal::h1c1398259f72ace6
                               at src/libstd/rt.rs:51
  26:     0x55eaf0ebe532 - main
  27:     0x7f19956b60b3 - __libc_start_main
  28:     0x55eaf0eae1be - _start
  29:                0x0 - <unknown>

**Comment by mitchmindtree** [comment]
Thanks for the issue!

It looks as though `wgpu` was unable to find a compatible graphics device. On Linux, wgpu uses Vulkan to detect and interface with graphics devices.

Is there any chance that you might need a vulkan driver update for your device following the Ubuntu update? Could you provide the output of `vulkaninfo` so that we may check that it is detected and see more details about your particular device? I can see that you have installed some nvidia drivers, but it can be useful to check `vulkaninfo` to make sure vulkan is working as expected.

If `vulkaninfo` does work and you do see your device, could you also try running the examples at [the wgpu-rs repository](https://github.com/gfx-rs/wgpu-rs)? This will help to determine whether or not the issue is related to the way we're using `wgpu`, or whether it is more related to `wgpu` itself.

**Comment by schwaaweb** [comment]
Thank you for your response. It appears that the nvidia-driver-440 version was not installed properly on Ubuntu-20.04 for me. 

When I tried to run nvidia-smi it indicated that it was unable to connect to the hardware. Another clue something was wrong, was that although I was getting video output to my screen I was not receiving the full 4K resolution of my monitor.

I downgraded the nvidia-driver-440 to nvidia-driver-435 and now nannou examples run on my system.

This was not a bug in nannou.cc but was a configuration problem on my end. Checking nvidia-smi showed me that the problem was on my end quickly.

Thank you again for your response, which pointed me in the right direction!
- Darwin
